### ISSUE TITLE
`Test` language kills the frontend (dev env only)

### ISSUE THREAD

**Issue by karwosts** [home-assistant/frontend#19171]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

In my frontend dev instance profile, I selected the `Test` language to see what it does. I immediately started seeing lots of errors, and I can no longer load any frontend pages. 

I get infinitely repeating errors like the following:

```
should-polyfill.js:51 Uncaught (in promise) RangeError: Incorrect locale information provided
    at Function.supportedLocalesOf (<anonymous>)
    at supportedLocalesOf (should-polyfill.js:51:30)
    at shouldPolyfill (should-polyfill.js:57:226)
    at polyfillIntl (intl-polyfill.ts:24:1)
    at eval (intl-polyfill.ts:62:1)
    at __webpack_require__.a (app.js:1899:12)
    at eval (intl-polyfill.ts:1:21)
    at ./src/resources/intl-polyfill.ts (src_common_datetime_use_am_pm_ts-src_common_entity_compute_state_display_ts-src_resources_int-4b7f76.js:119:1)
    at __webpack_require__ (app.js:1834:41)
    at eval (format_date.ts:17:82)
```

I can no longer load any pages at all. The only way out seems to be to shutdown the system and edit my language in .storage back to something else.

![image](https://github.com/home-assistant/frontend/assets/32912880/774e8ae9-45fd-4d67-b81d-0e46cf9e2273)

### Describe the behavior you expected

Anything but this 😂 

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.1.0b0

**Comment by bramkragten** [comment]
Ugh... 😄 maybe we should just kill the test language, it was to find untranslated text when only half of the UI was translated, but is less useful now...

**Comment by steverep** [comment]
Could probably fix just by giving it a real locale string of "en" instead of "test".  That said, I've never once used it, so I'm not opposed to just getting rid of it either.

**Comment by karwosts** [comment]
unstale

**Comment by steverep** [comment]
I can fix this pretty easily once #19988 is in.
